code/controllers/subsystems/supply.dm

```diff
@@ -50,8 +75,403 @@ var/datum/controller/subsystem/cargo/SScargo
 /datum/controller/subsystem/cargo/New()
 	NEW_SS_GLOBAL(SScargo)
 
-/datum/controller/subsystem/cargo/fire()
-	points += points_per_process
+
+/*
+	Loading Data 
+*/
+//Load the cargo data from SQL
+/datum/controller/subsystem/cargo/proc/load_from_sql()
+	if(!dbcon.IsConnected())
```

Comment:
Change to `if (!establish_db_connection(dbcon))`. That one should attempt a reconnect if unavailable as well, and is thus more reliable.